# Getting representation of French sentence with FlauBERT 
Let's explore how to use the pre-trained FlauBERT for obtaining the French sentence representation. First, let us import FlaubertTokenizer, FlaubertModel from transformers and also the torch library:


In [1]:
!pip install transformers==3.5.1

     |████████████████████████████████| 1.3MB 5.3MB/s 
     |████████████████████████████████| 2.9MB 15.4MB/s 
     |████████████████████████████████| 1.1MB 39.1MB/s 
     |████████████████████████████████| 890kB 45.5MB/s 
  Created wheel for sacremoses: filename=sacremoses-0.0.43-cp36-none-any.whl size=893261 sha256=63a06cab5f8874c6f42a90e2e5fc31207eba73fdb23a1820a7d18cbca581b5f6
  Stored in directory: /root/.cache/pip/wheels/29/3c/fd/7ce5c3f0666dab31a50123635e6fb5e19ceb42ce38d4e58f45
Successfully built sacremoses


In [2]:
from transformers import FlaubertTokenizer, FlaubertModel
import torch


Let's download and load the pre-trained flaubert_base_cased model: 

In [3]:
model = FlaubertModel.from_pretrained('flaubert/flaubert_base_cased')


Next, we download and load the tokenizer which is used for pre-training the  flaubert_base_cased model: 

In [4]:
tokenizer = FlaubertTokenizer.from_pretrained('flaubert/flaubert_base_cased')



Now, let us define the input sentence for which we want to compute the embedding: 

In [5]:
sentence = "Paris est ma ville préférée"


Next, we tokenize the sentence and get the token ids:

In [6]:
token_ids = tokenizer.encode(sentence)


Convert the tokens ids to torch tensors: 

In [7]:
token_ids = torch.tensor(token_ids).unsqueeze(0)


Obtain the representation of each token in our sentence using the pre-trained flaubert_base_casedmodel:


In [8]:
representation = model(token_ids)[0]



Let us check the shape of representation:

In [9]:
representation.shape

torch.Size([1, 7, 768])


As we can observe our representation size is [1, 7, 768] including the [CLS] and [SEP] tokens. We learned that the [CLS] token holds the aggregate representation of a sentence, so we get the representation of the [CLS] token and use it as the sentence representation: 


In [10]:
cls_rep = representation[:, 0, :]



That's it. In this way, we use the pre-trained FlauBERT and obtain the representation for French text. We can also use the pre-trained FlauBERT and fine-tune it for any downstream tasks. 